modules/core/04-channel/keeper/grpc_query_test.go

```diff
@@ -1219,6 +1229,45 @@ func (suite *KeeperTestSuite) TestQueryUnreceivedPackets() {
 			},
 			true,
 		},
+		{
+			"success basic unreceived packet commitments, ordered channel",
+			func() {
+				path := ibctesting.NewPath(suite.chainA, suite.chainB)
+				path.SetChannelOrdered()
+				suite.coordinator.Setup(path)
+
+				suite.chainA.App.GetIBCKeeper().ChannelKeeper.SetNextSequenceRecv(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 1)
+
+				expSeq = []uint64{1}
+				req = &types.QueryUnreceivedPacketsRequest{
+					PortId:                    path.EndpointA.ChannelConfig.PortID,
+					ChannelId:                 path.EndpointA.ChannelID,
+					PacketCommitmentSequences: []uint64{1},
+				}
+			},
+			true,
+		},
+		{
+			"success basic multiple unreceived packet commitments, ordered channel",
+			func() {
+				path := ibctesting.NewPath(suite.chainA, suite.chainB)
+				path.SetChannelOrdered()
+				suite.coordinator.Setup(path)
+
+				// Excersise scenario from issue 1532. NextSequenceRecv is 5, commitments for 2, 7, 9, 10
+				// are present. 2 is received so 7, 9, 10 should be unreceived.
+				expSeq = []uint64{7, 9, 10}
+				packetCommitments := []uint64{2, 7, 9, 10}
+				suite.chainA.App.GetIBCKeeper().ChannelKeeper.SetNextSequenceRecv(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, 5)
```

Comment:
maybe the `NextSequenceRecv` parameter can be set up as well generally as a variable which is then malleated per test?

Reply:
`expSeq` is already set up as so (If i'm understanding the comment correctly) and the rationale I got was that it's done since we need to assert its value against the expected in the main test body. Let me know if I got something wrong here.

Reply:
ah okay, i was just seeing the packet commitment sequences getting declared/set on the tests as well as the sequence number, so i was thinking to just set them as variables instead for readability -- but i am not very strongly opinionated, i agree that expSeq is the most important as we are testing for that value as the expected one. thanks!